server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/routes/JMAPApiRoutes.scala

```diff
@@ -103,12 +110,15 @@ class JMAPApiRoutes @Inject() (@Named(InjectionKeys.RFC_8621) val authenticator:
           ).`then`())
       )
 
-  private def processMethodWithMatchName(invocation: Invocation): SMono[Invocation] = invocation.methodName match {
-    case coreEcho.methodName => SMono.fromPublisher(coreEcho.process(invocation))
-    case _ => SMono.just(new Invocation(
-      MethodName("error"),
-      Arguments(Json.obj("type" -> "Not implemented")),
-      invocation.methodCallId))
+  private def processMethodWithMatchName(invocation: Invocation, mailboxSession: MailboxSession): SMono[Invocation] = {
+    SFlux.fromIterable(methods)
+      .filter(method => method.methodName == invocation.methodName)
```

Comment:
isn't what a Map is for?

Reply:
Not sure to get your comment. Here we want to filter out all the emthods except for the one passed as an argument in the request, so we can process it, not running all methods with a Map?